distribution/lib/Standard/AWS/0.0.0-dev/src/Database/Redshift/Internal/Redshift_Dialect.enso

```diff
@@ -66,6 +66,19 @@ type Redshift_Dialect
     generate_truncate_table_sql self table_name =
         Base_Generator.truncate_table_delete_from_style self table_name
 
+    ## PRIVATE
+       Generates SQL for limiting the number of rows using the `LIMIT` clause.
+    get_limit_part_sql : Integer -> SQL_Builder
+    get_limit_part_sql self limit =
+        SQL_Builder.code (" LIMIT " + limit.to_text)
+
+    ## PRIVATE
+       Generates SQL for limiting the number of rows using the `TOP` clause.
+    get_top_part_sql : Integer -> SQL_Builder
+    get_top_part_sql self limit =
+        _ = [limit]
+        SQL_Builder.code ""
```

Comment:
I don't like the fact that we have 2 functions here and one will always have to return empty code. IMO this is a bit of leaky abstraction - we assume there are exactly 2 ways to limit the query (probably true, I don't know a third way apart from TOP and LIMIT) - if we'd have to add another one we would need to modify all dialects again.

Could we merge this into a single method returning a pair - the SQL code and an `Integer` position where to put it - much like we do with the `context_extensions` mechanism that I introduced very recently? This would make it much more extensible.

Alternatively - if we actually want to assume there's only TOP and LIMIT - we could do a 'closed' approach - just do a simple getter on the dialect `query_limit_mode : Query_Limit_Mode` where `Query_Limit_Mode` would consist of 2 constructors: `Top` and `Limit`.

Reply:
Done